graylog2-server/src/main/java/org/graylog2/bootstrap/CmdLineTool.java

```diff
@@ -193,10 +197,16 @@ public void run() {
         final List<String> arguments = ManagementFactory.getRuntimeMXBean().getInputArguments();
         LOG.info("Running with JVM arguments: {}", Joiner.on(' ').join(arguments));
 
-        injector = setupInjector(configModule, pluginBindings, binder -> binder.bind(ChainingClassLoader.class).toInstance(chainingClassLoader));
+        configInjector = setupConfigInjector(configModule);
+        if (configInjector != null) {
+            final PluginBindings pluginBindings = new PluginBindings(plugins, configInjector);
+            injector = setupInjector(configModule, pluginBindings,
```

Comment:
Is it possible to use the `configInjector` as parent for the regular injector so we avoid duplicating the config instances?

Reply:
I like the idea and I tried it, but had some trouble with duplicate binding. I don't think it's worth the effort at the moment and we can still change it later.